### PR TITLE
converted metrics to coffee-script

### LINKED ISSUE #345: Convert api/metrics.coffee (small) to coffeescript
See epic here: https://github.com/jembi/openhim-core-js/issues/200

### PR DESCRIPTION
**Pull request by debonair**
@rcrichton please review closes #345

### CODE REVIEW

File: src/api/metrics.coffee [hunk 1]
```diff
@@ -11,70 +11,66 @@ metrics = require "../metrics"
...
-exports.getGlobalLoadTimeMetrics = `function *() {
-
-  var filtersObject = this.request.query;
-  var userRequesting = this.authenticated;
-
-  results = yield metrics.fetchGlobalLoadTimeMetrics(userRequesting, filtersObject);
+exports.getGlobalLoadTimeMetrics =  ->
+  filtersObject = this.request.query
+  userRequesting = this.authenticated
+  results = yield metrics.fetchGlobalLoadTimeMetrics(userRequesting, filtersObject)
```
Comment (rcrichton) [inline src/api/metrics.coffee]: You can remove the brackets from the function call

File: src/api/metrics.coffee [hunk 2]
```diff
@@ -11,70 +11,66 @@ metrics = require "../metrics"
...
-        this.body.push({
-          load: results[i].load,
-          avgResp: results[i].avgResp,
-          timestamp : moment.utc([results[i]._id.year, results[i]._id.month - 1, results[i]._id.day,results[i]._id.hour]).format()//, 'YYYY-MM-DD H+00:00'
-      });
-    }
-
-}`
+  i = 0
+  while i < results.length
```
Comment (rcrichton) [inline src/api/metrics.coffee]: For this you can do the following in coffeescript: `for result, i in results` to get access to `i` (coffeescript is cool :) )

File: src/api/metrics.coffee [hunk 3]
```diff
@@ -11,70 +11,66 @@ metrics = require "../metrics"
...
-exports.getGlobalStatusMetrics = `function *() {
-
-  var filtersObject = this.request.query;
-  var userRequesting = this.authenticated;
-
-  results = yield metrics.fetchGlobalStatusMetrics(userRequesting, filtersObject);
+exports.getGlobalStatusMetrics = ->
+  filtersObject = this.request.query
+  userRequesting = this.authenticated
+  results = yield metrics.fetchGlobalStatusMetrics(userRequesting, filtersObject)
```
Comment (rcrichton) [inline src/api/metrics.coffee]: You can remove the brackets here.

File: src/api/metrics.coffee [hunk 4]
```diff
@@ -11,70 +11,66 @@ metrics = require "../metrics"
...
-    this.body = results;
-  } else {
-    this.body = [];
-   //format the message to show what the console expects
-    for (var i = 0; i < results.length; i++) {
-      if (!results[i]._id.minute) {
+exports.getChannelMetrics = (time, channelId) ->
+  filtersObject = this.request.query
+  userRequesting = this.authenticated
+  results = yield metrics.fetchChannelMetrics(time, channelId, userRequesting, filtersObject)
```
Comment (rcrichton) [inline src/api/metrics.coffee]: Brackets

File: src/api/metrics.coffee [hunk 5]
```diff
@@ -11,70 +11,66 @@ metrics = require "../metrics"
...
+  if time == 'status'
```
Comment (rcrichton) [inline src/api/metrics.coffee]: You can `is` instead of `==`

File: src/api/metrics.coffee [hunk 6]
```diff
@@ -11,70 +11,66 @@ metrics = require "../metrics"
...
+    this.body = results
+  else
+    this.body = []
+    #format the message to show what the console expects
+    i = 0
+    while i < results.length
```
Comment (rcrichton) [inline src/api/metrics.coffee]: Same as above

**Comment by rcrichton** [comment]
Hey @debonair looks good, just a few comments.

**Comment by debonair** [comment]
@rcrichton I have made the changes.

**Comment by rcrichton** [comment]
Cool  @debonair looks good! :+1:

**Pull request merged**